### PR TITLE
Support archived or current mirrors by setting a variable

### PR DESCRIPTION
**Pull request by vincepii**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -14,11 +14,19 @@ kafka_scala_version: "2.11"
 kafka_java_enabled: true
 # set false if the role should not handle the java dependency
 
-kafka_mirror: http://apache.mirrors.tds.net/kafka
-kafka_url: "{{ kafka_mirror }}/{{ kafka_version }}/kafka_{{ kafka_scala_version }}-{{ kafka_version }}.tgz"
+# Use this selector to use a mirror for a "current" version or an "archived" one
+kafka_mirror_selector: "current"
+kafka_mirror:
+  current: http://apache.mirrors.tds.net/kafka
```
Comment (bbossy) [inline defaults/main.yml:20]: Seems like overkill for this: https://stackoverflow.com/questions/38592289/downloading-file-from-the-right-apache-mirror-with-wget

Just FYI.

**Review by bbossy — APPROVED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (vincepii) [reply defaults/main.yml:20]: It seems that `https://archive.apache.org/dist/kafka/` has everything actually.

**Review by chicofranchico — APPROVED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (vincepii) [reply defaults/main.yml:20]:
```
kafka_mirror: https://www.apache.org/dyn/closer.cgi?action=download&filename=/kafka
```
Would work for the tar.gz, but unfortunately it doesn't have the signatures.

Comment (bbossy) [reply defaults/main.yml:20]: Makes sense. Sigs should be pulled from a trusted source only.

**Pull request merged**